ted2zim/scraper.py

```diff
@@ -344,7 +344,9 @@ def generate_subtitle_list(self, video_id, langs, page_lang, audio_lang):
         """ List of all subtitle languages with link to their pages """
 
         subtitles = []
-        if self.subtitles_setting == ALL or (not self.source_languages and self.topics):
+        if self.subtitles_setting == ALL or (
```

Comment:
what are you fixing here?

Reply:
If you have not specified languages but have specified subtitles = none, it would give all subtitles. Though this worked, this should be changed to `and self.subtitles_setting != NONE` instead of `and not self.subtitles_setting`